apps/lib/http_server.c

```diff
@@ -225,6 +229,22 @@ BIO *http_server_init_bio(const char *prog, const char *port)
         goto err;
     }
 
+    /* Report back what address and port are used */
```

Comment:
Please make an aux function out of that, rather than copying code from `s_socket.c`, and use it in both places.

Reply:
BTW, nice trick to use `BIO_set_accept_port(acbio, port)` with `port` being 0 to auto-select any available local port. Please document this special feature in `doc/man3/BIO_s_accept.pod`.

Reply:
That nice trick should be common knowledge...  I'm again surprised how many do not know of this.
I can recommend "UNIX Network Programming" by Richard W. Stevens

Reply:
My knowledge of network programming is unfortunately very spotty.
Anyway, `BIO_set_accept_port()` is defined by OpenSSL, so its features should be documented there.
Also because this feature hopefully is usable also on non-UNIX systems ;-)

Reply:
Done with the aux function.  Good idea

Reply:
I'm proposing a respective doc extension in #15354.